python/finufft/pyproject.toml

```diff
@@ -72,3 +77,14 @@ before-build = "pip install delvewheel"
 # CIBW doesn't do vendoring of DLLs on Windows by default, so we have to
 # install delvewheel and run it.
 repair-wheel-command = "delvewheel repair -v --analyze-existing -w {dest_dir} {wheel}"
+# Need to repeat this here as the platform-specific config-settings overwrite
+# the general ones.
+# The last setting is required to force cmake to avoid using MSVC (the
+# default). By setting the generator to Ninja, cmake will pick gcc (mingw64) as
+# the compiler.
+config-settings = {"cmake.define.FINUFFT_ARCH_FLAGS" = "-march=x86-64", "cmake.define.CMAKE_VERBOSE_MAKEFILE" = "ON", "cmake.args" = "-G Ninja"}
+
+[[tool.cibuildwheel.overrides]]
+select = "*arm64*"
+# Don't enforce x84_64 when building on arm64.
+config-settings = {"cmake.define.FINUFFT_ARCH_FLAGS" = "", "cmake.define.CMAKE_VERBOSE_MAKEFILE" = "ON"}
```

Comment:
here we can still use march=native no? AFAIK from M1 to M3 they all support NEON only.

Reply:
I'd like to see this tested on those machines before enabling it. Maybe for 2.4?

Reply:
Okay, I compiled locally on my machine (apple-m1) with:
`cmake ../ -DFINUFFT_ARCH_FLAGS=-mcpu=apple-m3 -DFINUFFT_USE_OPENMP=OFF -DFINUFFT_BUILD_TESTS=ON -DCMAKE_C_COMPILER=gcc-14 -DCMAKE_CXX_COMPILER=g++-14`

and it passes the tests.

Reply:
Ok will add that, then.

Reply:
Done.

Reply:
Does this address a fix for 2.3.0, ie should it come in now and go in the release?

Reply:
Yes. The macOS x86_64 wheels were still crashing under Rosetta 2 (emulation mode for ARM chips), so we had to fix some things.